### ISSUE TITLE
Cannot create Index Patterns of remote clusters [Open Distro 1.12]

### ISSUE THREAD

**Issue by Phandora** [opensearch-project/security-dashboards-plugin#688]
We are facing some problems **creating index patterns of remote clusters in the 1.12 version**. 
(In Open Distro 1.11 version it works fine)

When trying to create the index pattern of a remote cluster we got the error `No matching indices found`. However, **the indexes do exist**. We will use the index pattern `*:*` to provide an example (The remote index pattern fails for every index pattern we attempt to create, even for sample indices):

![image](https://user-images.githubusercontent.com/33484080/105987240-50b3ec00-609e-11eb-8ae5-42de9645b8c8.png)

In step two, we can see there is something wrong before clicking on `Create index pattern` since the **Time field menu** does not appear, even when there are indices with timestamps:

![image](https://user-images.githubusercontent.com/33484080/105987457-983a7800-609e-11eb-9cd2-de8ff012544e.png)

We got the following error after creating the index pattern:

![image](https://user-images.githubusercontent.com/33484080/105987525-adafa200-609e-11eb-9216-55021acd6eb4.png)

Cross cluster searches work fine. If we go to `Discover`, the documents of remote clusters are displayed, but since there are no fields in the index pattern, **it does not allow filtering** using the document's fields.

![image](https://user-images.githubusercontent.com/33484080/105987805-1565ed00-609f-11eb-8adc-3ecd0d2bfc82.png)

If we create the index pattern using the API and adding some initial fields, they do appear on `Discover` and we can filter by them, but the index pattern can not be refreshed (we got the same error `No matching indices found`) so new fields can not be added to the index pattern.

**It is not a permissions issue**. We are using the default admin user ones:

```
  index_permissions:
    - index_patterns:
      - "*"
      allowed_actions:
      - "*" 
  tenant_permissions:
    - tenant_patterns:
      - "*"
      allowed_actions:
      - kibana_all_write
  cluster_permissions: 
    - "*"
 ```

We have tested it in Elastic 7.10.0 version (oss build) and the index patterns for remote clusters are created and working as usual. **The issue only applies to Open Distro 1.12 version** (As I said, it works in the Open Distro 1.11 version):

- When creating the index patterns of remote clusters, **the fields of the indices that match the index pattern are not included in the index pattern**. We got the error `No matching indices found`, even when the indices do exist and match.

- The index patterns of remote clusters **can not be refreshed**. Again, we got the error `No matching indices found`, even when the indices do exist and match.

- If we go to `Discover` and select the created index pattern, the documents from remote clusters are displayed, but since there are no fields in the index pattern, **it does not allow filtering** using the document’s fields.

## Steps to reproduce the issue

1. Have a remote cluster
2. Add remote cluster to local cluster
3. Insert data in the remote cluster, for instance:

```
curl -XPUT -k -H 'Content-Type: application/json' -u 'admin:admin' https://localhost:9200/books/_doc/1 -d '{"Dracula": "Bram Stoker"}'
```
4. Attempt to create an index pattern of the remote cluster from the local cluster (no remote one) in Kibana:

![image](https://user-images.githubusercontent.com/33484080/105988914-a5f0fd00-60a0-11eb-8787-c90950e4361e.png)

We got the error `No matching indices found`.

Also discussed on https://discuss.opendistrocommunity.dev/t/cannot-create-index-patterns-of-remote-clusters-open-distro-1-12-version/4732/3

Best regards,
Mayte Ariza

**Comment by AlexanderRasmussen** [comment]
We are having the exact same problem, but we dont have any remote cluster.

I believe you can find the root cause to your problem by looking at the security-auditlog index. 

We get errors saying: _**"MISSING_PRIVILEGES"**_ and the missing privilege seems to be **_"indices:admin/resolve/index"_**, you can try to look if you have the same problem.

The problem is there is not really any documentation on "indices:admin/resolve/index" and you can't see it in kibana if you try to add it manually.

**Comment by CodingBeagle** [comment]
We have the exact same issue.

And as you say @AlexanderRasmussen, it is because of the missing privilege. 

No documentation at all on the permission.

What's up with that?

**Comment by Phandora** [comment]
We have added the **indices:admin/resolve/index** permissions, but unfortunately, the issue is still going on:

```
  index_permissions:
  - index_patterns:
    - "*"
    allowed_actions:
    - "*"
    - "indices:admin/resolve/index"
  tenant_permissions:
  - tenant_patterns:
    - "*"
    allowed_actions:
    - "kibana_all_write"
  cluster_permissions:
  - "*"
  - "indices:admin/resolve/index"
```

Also, we did not find any error logs in the `security-auditlog` index.

**Comment by SitoRBJ** [comment]
Hello here!

I'm having the same problem as @Phandora, I have a remote odfe cluster attached to my main odfe cluster and I can't reload the indexes because I get the same error. 

I had to create the index pattern with the Kibana API, since it was impossible to create it from the interface. 

![image](https://user-images.githubusercontent.com/8913336/106136223-4b6fa380-6169-11eb-8c05-b32faaa8ff4f.png)

Any quick solution for this? I'm using Docker images from 1.12.0.

Regards

**Comment by Phandora** [comment]
Any update on this? We need a workaround as soon as possible since the issue is affecting our production environment.

We can provide any necessary assistance to debug the problem.

Best regards.

**Comment by Phandora** [comment]
Hey guys, any update on this?

I think the problem is detailed enough to reproduce it.  We can provide any necessary assistance. 
We need to fix it ASAP.

Regards

**Comment by ronansalmon** [comment]
Hi, 
We are having this issue as well with version 1.12.0
Steps to reproduce : 
From kibana
- Step 1 of 2: Define an index pattern
  - Create index pattern *:index-bla*,index-bla* (some index match)
  - Click next
- Step 2 of 2: Configure settings
  - Message : The indices which match this index pattern don't contain any time fields.
  - Time filed selector does not appear
  - Click create

-> No matching indices found: No indices match pattern *:index-bla*,index-bla*, Error fetching fields for index pattern  *:index-bla*,index-bla*

**Comment by ronansalmon** [comment]
Can this be related to https://github.com/opendistro-for-elasticsearch/security/issues/1032 ?

**Comment by Phandora** [comment]
This issue seems to be also related: https://discuss.opendistrocommunity.dev/t/cross-cluster-search-with-secured-clusters/5162.

They said it fails on `1.12.0`, `1.13.0` and `1.13.1` 🤕

**Comment by JustinasKO** [comment]
Any new here? having exactly same problem.

**Comment by davidlago** [comment]
Resolved in 1.13.2 and not present in newer OpenSearch versions. Please re-open if this is still happening and we can take a look. Thanks!
